Sign_Up_Demo/Controllers/SignUpViewController.swift

```diff
@@ -0,0 +1,54 @@
+import UIKit
+class SignUpViewController: UIViewController {
+    // MARK: Outlets
+    @IBOutlet weak var SignUpView: SignUpView!
+    // MARK: Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+}
+// MARK: Extention
+extension SignUpViewController {
+    // MARK: Actions Methods
+    @IBAction
+    func buttonSignUp(_ sender: UIButton) {
+        guard let userData = validateData() else { return }
+        let user = User(firstName: userData.firstName, lastName: userData.lastName, email: userData.email, password: userData.password, phoneNmbr: userData.phoneNumber)
+        navigateToUserDetailScreen(userDetail: user)
+    }
+    // MARK: Private Methods
+    private func validateData() -> (firstName: String, lastName: String, email: String, password: String, phoneNumber: String)? {
+        let firstName = SignUpView.txfFirstName.text ?? ""
+        let lastName = SignUpView.txfLastName.text ?? ""
+        let email = SignUpView.txfEmail.text ?? ""
+        let password = SignUpView.txfPassword.text ?? ""
+        let phoneNumber = SignUpView.txfPhone.text ?? ""
```

Comment:
And this is not the responsibility of validate data method. You can have a User object as an instance property of the SignupViewController and populate the user on in signup action method. This method will only validates that object and returns true or false. Or if you also want to return failure message you can use enum with 2 cases  validate and invalidate(message: String) and return one of the case depending on data.